src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1614,7 +1615,9 @@ public EmailWrapper sendConfirmationEmailForSubmission(String courseId, String f
         Assumption.assertFalse(student == null && instructor == null);
         
         try {
-            String timestamp = TimeHelper.formatTime12H(Calendar.getInstance().getTime());
+            String timestamp = TimeHelper.formatTime12H(
+                    TimeHelper.convertToUserTimeZone(Calendar.getInstance(TimeZone.getTimeZone("UTC")),
+                            session.getTimeZone()).getTime());
```

Comment:
The issue is `timestamp` is being passed as `String`. If you can modify such that the timestamp is passed as a `Calendar` or `Date`, you can perform this conversion at the `EmailGenerator` class, thus making the operation testable.